src/commons.ts

```diff
@@ -106,47 +108,15 @@ export default class Commons {
     }
   }
 
-  public async InitalizeSettings(
-    askToken: boolean,
-    askGist: boolean
-  ): Promise<LocalConfig> {
+  public async InitalizeSettings(): Promise<LocalConfig> {
     const settings = new LocalConfig();
     const extSettings = this.GetSettings();
     const cusSettings = await this.GetCustomSettings();
 
-    if (cusSettings.token === "") {
-      if (askToken === true) {
-        askToken = !cusSettings.downloadPublicGist;
-      }
-
-      if (askToken) {
-        if (cusSettings.openTokenLink) {
-          vscode.commands.executeCommand(
-            "vscode.open",
-            vscode.Uri.parse("https://github.com/settings/tokens")
-          );
-        }
-        const tokTemp: string = await this.GetTokenAndSave(cusSettings);
-        if (!tokTemp) {
-          const msg = localize("common.error.tokenNotSave");
-          vscode.window.showErrorMessage(msg);
-          throw new Error(msg);
-        }
-        cusSettings.token = tokTemp;
-      }
+    if (cusSettings.token === "" || extSettings.gist === "") {

```

Comment:
If you explore the process of downloading public gist of another user. 
It doesnt ask the token for downloading.
So need to check if the public gist is enabled it should not check for token.

We need to make it supported

Reply:
Ok, my new commit supports public gists